mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
and by "associate the term", I mean associating the term workspace with, for example 'plan' in fftw